src/ensRegistry.ts

```diff
@@ -46,16 +49,50 @@ function makeSubnode(event:NewOwnerEvent): string {
   return crypto.keccak256(concat(event.params.node, event.params.label)).toHexString()
 }
 
+function recurseDomainDelete(domain: Domain): string {
+  if (domain.parent !== null) {
+    const parentDomain = Domain.load(domain.parent!)
+    let resolverIsNull = false
+    if (domain.resolver !== null) {
```

Comment:
The resolver's ID is the concatenation of the domain id and the resolver address - so you should be able to construct the ID of a zero resolver and compare it here, to avoid the need to load the object.